configurationproperties-annotationprocessor/src/main/java/com/vaadin/flow/spring/ConfigurationPropertiesAnnotationProcessor.java

```diff
@@ -0,0 +1,386 @@
+        return fieldDeclaration.getVariables().size() > 0
+                && fieldDeclaration.getVariable(0) != null
+                && fieldDeclaration.getVariable(0).
+                    getInitializer().isPresent();
+    }
+
+
+    private void writeJavaCodeToFile(String annotatedClassName,
+                                     String generatedClassName,
+                                     String prefix,
+                                     List<FieldDescriptor> fields)
+            throws IOException {
+
+        JavaFileObject builderFile = processingEnv.getFiler()
+                .createSourceFile(generatedClassName);
+        try (PrintWriter out = new PrintWriter(builderFile.openWriter())) {
+
+            String packageName = generatedClassName
+                    .substring(0, generatedClassName.lastIndexOf("."));
+            String simpleClassName = generatedClassName
+                    .substring(packageName.length() + 1);
+
+            out.println("package " + packageName + ";");
+            out.println();
+
+            out.println(getImportsAsString(annotatedClassName));
+
+            out.println("import javax.annotation.PostConstruct;");
+            out.println("import org.springframework.boot.context.properties." +
+                    "ConfigurationProperties;");
+            out.println();
+            out.println();
+
+            out.println("/**");
+            out.println("* generated by " + getClass().getName());
+            out.println("* from " + annotatedClassName);
+            out.println("*/");
+            out.println("@ConfigurationProperties(prefix = \"" +
+                    prefix + "\")");
+            out.println("public class " + simpleClassName + " {");
+
+            fields.forEach(f -> writeField(out, f));
+
+            writePostConstruct(out, fields);
+
+            out.println("}");
+        }
+    }
+
+    private String getImportsAsString(String className) {
+        StringBuilder value = new StringBuilder();
+        CompilationUnit cu = getCompilationUnit(className);
+        if (cu != null) {
+            cu.accept(new VoidVisitorAdapter<StringBuilder>() {
+
+                @Override
+                public void visit(ImportDeclaration fieldDeclaration,
+                                  StringBuilder arg) {
+                    super.visit(fieldDeclaration, arg);
+                    value.append(fieldDeclaration.toString());
+                }
+
+            }, value);
+        }
+        return value.toString();
+    }
+
+    private void writePostConstruct(PrintWriter out,
+                                    List<FieldDescriptor> fields) {
+        out.println(" @PostConstruct");
+        out.println(" public void transferValuesToSystem() {");
+        fields.forEach(f -> {
+            out.println("  if (" + f.fieldName + " != null) " +
+                    "System.setProperty(\"vaadin." + f.formerPropertyName
```

Comment:
In fact the generated configuration class itself is not used anywhere but in `SpringBootAutoConfiguration`, and only for the `urlMapping` parameter. All of the other values inside the configuration must be sent to the vaadin servlet somehow. 

The vaadin servlet reads these values via the `DeplymentConfiguration` class, which reads the values from web.xml or from the system properties. Hence setting system.properties.

Anyway, I chose this option as I understood it was less intrusive (less original code modified). The other option I managed was to serialize all the values to a Map, so it could be passed to the servlet as the initial parameters inside the `SpringBootAutoConfiguration::servletRegistrationBean` method.

Reply:
This is totally different issue.

And it's already exists: https://github.com/vaadin/spring/issues/478 (or almost the same).

This is another aspect of Spring: it doesn't map `Environment` properties and properties class but I would expected this:
- _IF_ you have `application.properties` and you have a config properties bean then the bean will be filled with the values from the properties
- BUT if you use the config properties bean to modify the values then they stay modified inside the bean only without env properties modifications.
https://docs.spring.io/spring-boot/docs/current/reference/html/features.html#features.external-config

For me this is a bug since I have to care about two sources of authority in my code: `env` properties and Java bean properties. 

Let's don't mix the different issues together: it's not necessary to generate the Java class and produce an issue with it's usage.
You may generate a Json file instead and forget about all Java related issues in Spring.

If you want to have this kind of functionality then you may use `Environment` bean to set properties into it instead of `System`.
`System` properties should not be touched at all : this is incorrect.

Reply:
Sorry Denis, I have seen your comment now.

I used system properties as I was not sure that, if I set them using `Environment`, they would be actually read by the vaadin servlet.

I chose a java class instead of a json file as I thought that the relaxed naming scheme (which is one of the requirements) was not going to work with a json file.

Regarding the system properties, another option would be to serialize the generated `VaadinConfigurationProperties` object to a properties map, and pass it to `ServletRegistrationBean::setInitparameters` inside  `SpringBootAutoConfiguration::servletRegistrationBean`.

Please note that I need this serialization to warranty that properties are passed using the exact names which the vaadin servlet is expecting.